ga4gh/datamodel/__init__.py

```diff
@@ -26,6 +27,59 @@ def _cleanupHtslibsMess(indexDir):
         shutil.rmtree(indexDir)
 
 
+class PysamFileHandleCache(object):
+    """
+    Cache for opened file handles We use a deque which has the
+    advantage to have push/pop operations in O(1) We always add
+    elements on the left of the deque and pop elements from the right.
+    When a file is accessed, its priority gets updated, it is put at
+    the "top" of the deque.
+    """
+
+    def __init__(self):
+        self._cache = collections.deque()
+        self._memoTable = dict()
+        # Initialize the value even if it will be set up by the config
+        self._maxCacheSize = 50
+
+    def setMaxCacheSize(self, size):
+        self._maxCacheSize = size
+
+    def add(self, dataFile, handle):
+        self._cache.appendleft((dataFile, handle))
+
+    def update(self, dataFile, handle):
+        self._cache.remove((dataFile, handle))
+        self.add(dataFile, handle)
+
+    def removeLRU(self):
```

Comment:
Two things here:

1. Does adding this method make the `getFileHandle` method easier to understand? I would say, probably not.
2. If we do keep the method, we should change it to `removeLru` for consistency with existing conventions.